sys/dev/isp/isp.c

```diff
@@ -4364,15 +4726,14 @@ isp_read_nvram_2400(ispsoftc_t *isp)
 static void
 isp_rd_2400_nvram(ispsoftc_t *isp, uint32_t addr, uint32_t *rp)
 {
+	fcparam *fcp = FCPARAM(isp, 0);
 	int loops = 0;
-	uint32_t base = 0x7ffe0000;
+	uint32_t base; // = 0x7ffe0000;
 	uint32_t tmp = 0;
 
-	if (IS_26XX(isp)) {
-		base = 0x7fe7c000;	/* XXX: Observation, may be wrong. */
-	} else if (IS_25XX(isp)) {
-		base = 0x7ff00000 | 0x48000;
-	}
+	base = fcp->flash_data_addr + addr;
+	addr = 0;
+
 	ISP_WRITE(isp, BIU2400_FLASH_ADDR, base | addr);
```

Comment:
I am not getting idea of first clearing the addr and then or'ing it to base.

Reply:
I've overseen this, thanks for the hint